### PR TITLE
Update to App Store release notes

### PR DESCRIPTION
**Pull request by guarani**
This PR makes some edits to the release notes for clarity. While I'm reasonably happy with the changes, please feel free to suggest further edits.

cc @mokagio 

## Regression Notes
1. Potential unintended areas of impact

Since this changes release notes after they've been through editorial, there's a risk of introducing errors. 

PR submission checklist:

- [ ] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Review by mokagio — APPROVED** [review]
Thanks @jkmassel @guarani 🙌

**Review by mokagio — DISMISSED** [review]
Woops. Only one of the `.po` `msgid` was fixed.

File: WordPress/Jetpack/Resources/AppStoreStrings.po [hunk 1]
```diff
@@ -15,7 +15,8 @@ msgstr ""
 
 #. translators: Subtitle to be displayed below the application name in the Apple App Store. Limit to 30 characters including spaces and commas!
 msgctxt "app_store_subtitle"
-msgid "Make your WordPress site faster and safer."
+msgid "Make your WordPress site faster and safer.
+"
```
Comment (mokagio) [inline WordPress/Jetpack/Resources/AppStoreStrings.po]:
```suggestion
msgid "Make your WordPress site faster and safer."
```

**Review by jkmassel — APPROVED** [review]
Nice job team!! 😁

**Review by mokagio — APPROVED** [review]

**Pull request merged**